bnelson777/id.ly
# IL-114: Add file system and file encryption

- AsyncStorage has been replaced with with files stored in a directory.
- All actions that modify cards and messages now use the new file system.
- Cards and messages are now encrypted when written and decrypted when read.
- A key and initialization vector is generated on starting the app for AES encryption. This key is stored in the device's secure keychain/keystore for future use.
- Unused functions such as Delete and Update card have been removed.

Testing procedure:

1. Clear all data.
2. Reload and observe encrypted card and message data in console log.
3. Through wallet, go to add a card. Observe encrypted card data in console log when card is added.
4. Create a new message. Observe encrypted message data in console log when message is created.
5. Perform any other actions that access messages and cards and ensure that the forward-facing behavior is visibly unchanged.
